x/fees/types/fee.go

```diff
@@ -25,12 +26,21 @@ func (i DevFeeInfo) Validate() error {
 		return err
 	}
 
+	if ethermint.IsZeroAddress(i.ContractAddress) {
+		return sdkerrors.Wrapf(
+			sdkerrors.ErrInvalidAddress, "address must not be empty %s",
+			i.ContractAddress,
+		)
+	}
+
 	if _, err := sdk.AccAddressFromBech32(i.DeployerAddress); err != nil {
 		return err
 	}
 
-	if _, err := sdk.AccAddressFromBech32(i.WithdrawAddress); err != nil {
-		return err
+	if i.WithdrawAddress != "" {
```

Comment:
Q1: are we fine with the deployer and withdraw addresses being set to the `ZeroAddress`?
Q2: are we fine the deployer and withdraw addresses being equal?

Reply:
> Q1: are we fine with the deployer and withdraw addresses being set to the ZeroAddress?
For the contract address, `common.Address{}` results in `0x0000000000000000000000000000000000000000`, so there was a chance of misuse. 
Deployer is also signer, so it should never be 0. Withdraw address can be empty.

> Q2: are we fine the deployer and withdraw addresses being equal?
Good question. But I would not handle this in validation. @fedekunze , if there is a registration with `deployer == withdraw`, should I just make `withdraw = ""`, without throwing an error?

Reply:
Q1: `0x0000000000000000000000000000000000000000` would fail since it's not a bech32 address, but we should error if the bech32 representation of the zero address is used

Q2: I think we can accept `deployer == withdraw` on the validation, but if that's the case, we shouldn't set the withdraw address to save some disk space.

Reply:
> we should error if the bech32 representation of the zero address is used
This is what I meant, apologies if it was unclear